main.go

```diff
@@ -48,6 +53,10 @@ func main() {
 		credentialsPath = fmt.Sprintf("%s/.aws/credentials", usr.HomeDir)
 	}
 
+	authCredentialsProvider := credentials.NewSharedCredentials(credentialsPath, authProfileFlag)
+	_, err := authCredentialsProvider.Get()
+	check(err)
+
 	credentialsProvider := credentials.NewSharedCredentials(credentialsPath, profileFlag)
 	creds, err := credentialsProvider.Get()
 	check(err)
```

Comment:
I may be wrong, but it seems like some of this is not necessary. Can't it be written like this?

```go
	credentialsProvider := credentials.NewSharedCredentials(credentialsPath, profileFlag)
	creds, err := credentialsProvider.Get()
	check(err)

	if authProfileFlag != "" {
		profileFlag = authProfileFlag
	}
```

And then remove this part:

```go
	if authProfileFlag == "" {
		authProfileFlag = profileFlag
	}
```

And remove the one line change below.

Reply:
So the one downside to that, and maybe it's fine, is that it doesn't verify (via `authCredentialsProvider.Get`) that the auth-profile credentials are valid.  I'll try it this way though and see if it works.

Reply:
Seems to work fine.  👍

Reply:
Ok, that's great! I think I understand the change much better now. Thanks for testing it.

What is left now is to update the Usage section in the README. Feel free to add text to explain how it can be used under the code block as well. It doesn't all have to be in the flag help string (but feel free to update that as well if you can think of a better string).

Reply:
Will do, and I'll send an update later.  Thanks!